### PR TITLE
V3.4.1

### PR DESCRIPTION
**Pull request by SC426**
[Upstream](https://github.com/apache/spark/tree/v3.4.1/python)

## Actions
- Updated version to 3.4.1
- Changed minimum `python` version to `3.8`
- Updated build script to include `--no-build-isolation`
- Dependency updates
- Removed outdated patch file `0001-Disable-symlinks-on-Windows`

### CODE REVIEW

**Review by sumit0190 — APPROVED** [review]
LGTM, but maybe we should open an issue/ticket for adding the Spark Connect dependencies listed here:
https://github.com/apache/spark/blob/6b1ff22dde1ead51cbf370be6e48a802daae58b6/python/setup.py#L314
We don't have to do it right now though, as that would require updating `grpcio` which might take a little while and this is urgent.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,38 +1,34 @@
...
     - python
-    - py4j ==0.10.9.3
+    - py4j ==0.10.9.7
     # extras_require dependencies
-    - numpy >=1.14
-    - pandas >=0.23.2
-    - pyarrow >=1.0.0
+    - numpy >=1.15
+    - pandas >=1.0.5
+    - pyarrow >=4.0.0
```
Comment (sumit0190) [inline recipe/meta.yaml]: Isn't the minimum version still `1.0.0`?
https://github.com/apache/spark/blob/6b1ff22dde1ead51cbf370be6e48a802daae58b6/python/setup.py#L134

Comment (SC426) [reply recipe/meta.yaml]: Yes, I'll fix that. Don't know what I was looking at.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,37 +1,32 @@
...
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 0b81359262ec6e9ac78c353344e7de026027d140c6def949ff0d80ab70f89a54
-  patches:
-    - 0001-Disable-symlinks-on-Windows.patch
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:12]: You can remove the file too.

**Pull request merged**